PYME/recipes/localisations.py

```diff
@@ -1030,8 +1030,74 @@ def run(self, inputName):
             cmorph.measure_3d(x, y, z, output=measurements[cluster_index])
 
             indi = indf
+        
+        # propagate the labelKey used to generate these measurements
+        output = tabular.MappingFilter(tabular.RecArraySource(measurements))
+        output.addColumn(self.labelKey, unique_labels)
```

Comment:
This looks wrong ... should be `np.arange(maxLabel)` rather than `unique_labels`

Reply:
@barentine - any thoughts of this

Reply:
Sorry for the delay! I think `unique_labels` is considerably safer here than a `np.arange` call. `unique_labels` matches what we actually iterate over, and is chosen to be robust against cases where you've filtered/merged/somehow ended up with non-contiguous labels, e.g. 0, 1, 2, 4 which would cause us to propagate the wrong labels if we used `np.arange`. Obviously nice if labels are contiguous, but would be a nasty place to silently break things if they aren't.

Reply:
Note comments in code above, which would imply `arange` is the correct choice:

``` python
# Note that missing labels result in zeroed entries (i.e. the initial values are not changed).
# Missing values can be filtered out later, if desired, by filtering on the 'counts' column, but having a dense
# array where index == label number makes any postprocessing in which we might want to find the data
# corresponding to a particular label MUCH easier and faster.

```

and the assignment:
 `output=measurements[cluster_index])`

We iterate over the set of unique labels, but assign into the array using the label, not it's position.

Reply:
Yep, looks good! Sorry I missed that and was thinking more aligns of the new `AddMeasurementsByLabel`. Please feel free to change to arange here!